### PR TITLE
SDKS-2780 Allow developer to customize the cookie header to the outgoing request from the SDK

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[SDKS-2780](https://bugster.forgerock.org/jira/browse/SDKS-2780)

# Description
SDKS-2780 Allow developer to customize the cookie header to the outgoing request from the SDK

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    private val singleSignOnManager: SingleSignOnManager
+    private val cacheRef = AtomicReference<MutableSet<Cookie>?>()
+    private val cacheIntervalMillis: Long
+    private val cookieMarshaller = CookieMarshaller()
+
+    init {
+        this.singleSignOnManager = singleSignOnManager ?: Config.getInstance().singleSignOnManager
+        this.cacheIntervalMillis = cacheIntervalMillis
+            ?: (context!!.resources.getInteger(R.integer.forgerock_cookie_cache) * 1000).toLong()
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt:30]: replace bang bang to optional
 
this.cacheIntervalMillis = cacheIntervalMillis
            ?: ((context?.resources?.getInteger(R.integer.forgerock_cookie_cache) ?: 0) * 1000).toLong()

File: forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            this.cacheIntervalMillis = cacheIntervalMillis
+            return this
+        }
+
+        fun build(): SecureCookieJar {
+            return SecureCookieJar(context, singleSignOnManager, cacheIntervalMillis)
+        }
+
+        override fun toString(): String {
+            return "SecureCookieJar.SecureCookieJarBuilder(context=" + context + ", singleSignOnManager=" + singleSignOnManager + ", cacheIntervalMillis=" + cacheIntervalMillis + ")"
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt:148]: may be more cleaner  

 return "SecureCookieJar.SecureCookieJarBuilder(context=$context, singleSignOnManager=$singleSignOnManager, cacheIntervalMillis=$cacheIntervalMillis)"

**Comment by jeyanthanperiyasamy** [comment]
implementation looks good , just minor comment

File: forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt]: I changed the context to non-null.

File: forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt (hunk 2 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/SecureCookieJar.kt]: Updated

**Review by jeyanthanperiyasamy — DISMISSED** [review]

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍🏻

**Pull request merged**